### PR TITLE
Backend Clients for Frontend

### PR DESCRIPTION
**Pull request by AinoraZ**
Added clients for each corresponding backend controller.

Note: I have made another task for switching out the current API calls with these clients. I'd rather merge this now and switch it out later, just so we could start using these clients for new code.

Unrelated fixes:
  - Added .vscode/settings.json to auto-fix eslint on save and a few other options.
  - Disabled rule for enforcing use of ```this``` in class methods.

### CODE REVIEW

**Comment by AinoraZ** [comment]
[Task 6: Create Frontend Clients For Accessing API](https://trello.com/c/dQPEobva/60-task-6-create-frontend-clients-for-accessing-api)

File: billchop-fe/src/backend/clients/BillClient.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import Axios, { AxiosResponse } from "axios";
+import url from "url-join";
+import BaseClient from "./AbstractClient";
+import Bill from "../models/Bill";
+import { CreateNewBill } from "../models/CreateNewBill";
+
+const groupParam = "groupId";
```
Comment (tmaciulis22) [inline billchop-fe/src/backend/clients/BillClient.ts]: Dunno if typescript has same convention as other languages, but shouldn't this be in all caps?

File: billchop-fe/src/components/GroupTable.tsx [hunk 2]
```diff
@@ -3,10 +3,10 @@ import Table from "react-bootstrap/Table";
...
 import Axios from "axios";
-import Group from "../api/Group";
+import Group from "../backend/models/Group";
 import BillSplitInput from "./BillSplitInput";
-import User, { CURRENT_USER_ID } from "../api/User";
-import Loan from "../api/Loan";
+import User, { CURRENT_USER_ID } from "../backend/models/User";
+import Loan from "../backend/models/Loan";
 
 const BASE_URL_API_GROUPS = "https://localhost:44333/api/groups/";
```
Comment (tmaciulis22) [inline billchop-fe/src/components/GroupTable.tsx]: When changing API calls don't forget to delete these consts

**Review by tmaciulis22 — DISMISSED** [review]
Very good job 👏

File: billchop-fe/src/backend/clients/BillClient.ts (hunk 1 above)
Comment (AinoraZ) [reply billchop-fe/src/backend/clients/BillClient.ts]: Yeah you are right, will change

File: billchop-fe/src/components/GroupTable.tsx (hunk 2 above)
Comment (AinoraZ) [reply billchop-fe/src/components/GroupTable.tsx]: Will do so when changing to clients in next PR. I'm pretty sure the linter would scream at me anyway for unused variables 😅.

**Review by tmaciulis22 — APPROVED** [review]

**Pull request merged**